lib/Migration/Version1120Date20220412111833.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Anna Larch <anna.larch@gmx.net>
+ *
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Type;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version1120Date20220412111833 extends SimpleMigrationStep {
+	/** @var IDBConnection $connection */
+	private $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		$schema = $schemaClosure();
+
+		// Remove old unnamed attachments FK
+		$attachmentsTable = $schema->getTable('mail_attachments');
+		$fks = $attachmentsTable->getForeignKeys();
+		foreach ($fks as $fk) {
+			$attachmentsTable->removeForeignKey($fk->getName());
+		}
+
+		// Remove old unnamed FK and change primary key type for recipients table
+		$recipientsTable = $schema->getTable('mail_recipients');
+		$fks = $recipientsTable->getForeignKeys();
+		foreach ($fks as $fk) {
+			$recipientsTable->removeForeignKey($fk->getName());
+		}
+
+		$recipientsTable->changeColumn('id', [
+			'type' => Type::getType('bigint'),
+			'length' => 20,
+		]);
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('mail_recipients');
+		$qb->execute();
+
+		// Truncate and change primary key type for messages table
+		$qb1 = $this->connection->getQueryBuilder();
+		$qb1->delete('mail_messages');
+		$qb1->execute();
+
+		$messagesTable = $schema->getTable('mail_messages');
+		$messagesTable->changeColumn('id', [
+			'type' => Type::getType('bigint'),
+			'length' => 20,
+		]);
```

Comment:
Does not have effect here, same as for the col change on recipientsTable, or am i missing something here? Unless the changes are being kept by Doctrine across methods. Then I said nothing,